src/main/java/org/opentripplanner/routing/core/State.java

```diff
@@ -104,21 +104,21 @@ public State(Vertex vertex, Edge backEdge, long timeSeconds, long startTime, Rou
         // this should be harmless since reversed clones are only used when routing has finished
         this.stateData.opt = options;
         this.stateData.startTime = startTime;
-        this.stateData.usingRentedBike = false;
+        this.stateData.bikeRentalState = BikeRentalState.BEFORE_RENTING;
```

Comment:
This won't properly account for arriveBy searches where the user has dropped off a bike at the chronological end (therefore, first arriveBy state) of a journey. In our fork we added code that would equivalently set this BikeRentalState to `RENTING_FLOATING` if the first seen street edge had conditions that would allow a floating rental to end there. See [our code here](https://github.com/ibi-group/OpenTripPlanner/blob/65dcf0a4142e31028cf9d1b2c15ad32dd1084252/src/main/java/org/opentripplanner/routing/core/State.java#L176-L198).